### ISSUE TITLE
isClearable button is missing aria-label

### ISSUE THREAD

**Issue by JamesCyrilJordan** [Hacker0x01/react-datepicker#1282]
### Expected behavior
When visually impaired individual tabs focus to "clear" button, they should hear a message indicating that the button will clear the current field.

### Actual behavior
When visually impaired individual tabs focus to "clear" button, they hear the message "number".

### Steps to reprodcue
Using Jaws or some other 508 compliant screen reader. Tab through to a date field with a valid date, and focus on the "clear" x button. The screen reader will read the word "number" to you.
